File: mobile-android/app/src/main/java/com/erenlermarket/app/domain/repository/CommerceRepositories.kt
package com.erenlermarket.app.domain.repository

import com.erenlermarket.app.domain.model.Address
import com.erenlermarket.app.domain.model.CartItem
import com.erenlermarket.app.domain.model.CheckoutPreview
import com.erenlermarket.app.domain.model.NewAddress
import com.erenlermarket.app.domain.model.Order
import com.erenlermarket.app.domain.model.PlaceOrderInput
import com.erenlermarket.app.domain.model.Product

interface CartRepository {
    suspend fun cart(): List<CartItem>
    suspend fun addItem(productId: String, quantity: Int)
    suspend fun updateQuantity(productId: String, quantity: Int)
    suspend fun removeItem(productId: String)
    suspend fun clear()
    suspend fun checkoutPreview(couponCode: String?): CheckoutPreview
}

interface WishlistRepository {
    suspend fun wishlist(): List<Product>
    suspend fun add(productId: String)
    suspend fun remove(productId: String)
}

interface AddressRepository {
    suspend fun addresses(): List<Address>
    suspend fun create(address: NewAddress): Address
    suspend fun delete(id: String)
}

interface OrderRepository {
    suspend fun orders(): List<Order>
    suspend fun order(id: String): Order
    suspend fun place(input: PlaceOrderInput): Order
    suspend fun cancel(id: String, reason: String?): Order
}

File: mobile-ios/Modules/Domain/Sources/Repositories/CommerceRepositories.swift
/// Sepet (tümü sunucu tarafında).
public protocol CartRepository: Sendable {
    func cart(language: String) async throws -> [CartItem]
    func addItem(productID: String, quantity: Int) async throws
    func updateQuantity(productID: String, quantity: Int) async throws
    func removeItem(productID: String) async throws
    func clear() async throws
    func checkoutPreview(
    couponCode: String?,
    addressID: String?,
    language: String
  ) async throws -> CheckoutPreview
}

/// İstek listesi.
public protocol WishlistRepository: Sendable {
    func wishlist(language: String) async throws -> [Product]
    func add(productID: String) async throws
    func remove(productID: String) async throws
}

public protocol AddressRepository: Sendable {
    func addresses() async throws -> [Address]
    func create(_ address: NewAddress) async throws -> Address
    func delete(id: String) async throws
}

public protocol OrderRepository: Sendable {
    func orders(language: String) async throws -> [Order]
    func order(id: String, language: String) async throws -> Order
    func place(_ input: PlaceOrderInput, language: String) async throws -> Order
    func cancel(id: String, reason: String?, language: String) async throws -> Order
}
